src/summary.rs

```diff
@@ -40,39 +40,114 @@ impl std::fmt::Display for Rewards {
     }
 }
 
-/// Struct for holding the information of what to be displayed with the `info`
-/// command
-#[derive(Deserialize, Serialize, Debug)]
+/// struct for flexibly updating the fields of the summary
+#[derive(Default, Debug)]
+pub(crate) struct SummaryUpdateFields {
+    pub(crate) is_plotting_finished: bool,
+    pub(crate) is_new_block_farmed: bool,
+    pub(crate) is_new_vote: bool,
+    pub(crate) maybe_new_reward: Option<Rewards>,
+    pub(crate) maybe_last_block: Option<u64>,
```

Comment:
I think you should cache these 4 things:
- last seen finalized block
- total votes
- total farmed blocks
- total rewards

Reply:
but I do require `is_plotting_finished` as well